### PR TITLE
statistics: fix the unexpected estimation error on full sampling

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve?

Problem Summary:

This pr fixes the following problems for full sampling
- When init the stats, the topn cannot be inited since there's no cmsketch.
- When checking `OutOfRang`, doesn't check whether the value is in TopN, it will go wrong.
- When loading stats from json, the topn is not loaded since the cmsketch is nil.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

Fix the three problems.

### Check List 

Tests 

- Unit test(to be added)

### Release note 

- No release note

### CODE REVIEW

File: statistics/cmsketch.go [hunk 1]
```diff
@@ -386,8 +386,9 @@ func CMSketchToProto(c *CMSketch, topn *TopN) *tipb.CMSketch {
 
 // CMSketchAndTopNFromProto converts CMSketch and TopN from its protobuf representation.
 func CMSketchAndTopNFromProto(protoSketch *tipb.CMSketch) (*CMSketch, *TopN) {
+	retTopN := TopNFromProto(protoSketch.TopN)
```
Comment (time-and-fate) [inline statistics/cmsketch.go:389]: static check says need a nil check here.

File: statistics/histogram.go [hunk 2]
```diff
@@ -1504,13 +1504,14 @@ func (coll *HistColl) NewHistCollBySelectivity(sc *stmtctx.StatementContext, sta
 }
 
 func (idx *Index) outOfRange(val types.Datum) bool {
-	if idx.Histogram.Len() == 0 {
+	if idx.Histogram.Len() == 0 && len(idx.TopN.TopN) == 0 {
```
Comment (time-and-fate) [inline statistics/histogram.go:1507]: also need a nil check here.

**Review by eurekaka — APPROVED** [review]
/lgtm

**Review by time-and-fate — APPROVED** [review]
/lgtm

**Comment by time-and-fate** [comment]
Need to add comments for the newly added `(c *TopN).Num` method.

**Pull request merged**